indexd/index/drivers/alchemy.py

```diff
@@ -300,6 +334,85 @@ def delete(self, did, rev):
 
             session.delete(record)
 
+    def get_all_versions(self,baseid):
+        '''
+        Get all records with baseid
+        '''
+
+        ret = dict()
+        with self.session as session:
+            query = session.query(IndexRecord)
+            records = query.filter(IndexRecord.baseid == baseid).all()
+
+            if(len(records) == 0):
+                raise NoRecordFound('no record found')
+
+            for idx, record in enumerate(records):    
+                rev = record.rev
+                did = record.did
+
+                form = record.form
+                size = record.size
+
+                urls = [u.url for u in record.urls]
+                hashes = {h.hash_type: h.hash_value for h in record.hashes}
+
+                updated_last_by = record.updated_last_by
+
+                ret[idx] = {
+                    'did': did,
+                    'rev': rev,
+                    'size': size,
+                    'urls': urls,
+                    'hashes': hashes,
+                    'form': form,
+                    'updated_last_by': updated_last_by,
+
+                }
+
+        return ret
+
+    def get_latest_version(self,baseid):
+        '''
+        Get all records with baseid
+        '''
+        ret = {};
+        with self.session as session:
+            query = session.query(IndexRecord)
+            records = query.filter(IndexRecord.baseid == baseid).order_by(IndexRecord.updated_last_by).all()
+
+            if(len(records) == 0):
```

Comment:
`if not records` is slightly more idiomatic Python, and in cases where the list could be large saves the time of counting the exact length.